src/components/ui/button.tsx

```
@@ -33,27 +33,40 @@ const buttonVariants = cva(
       {...props}
-    />
-  )
+    >
+      {loading && <Loader2 className="animate-spin" />}
+      {loading ? loadingText : children}
+    </Comp>
+  );
 }
 
-export { Button, buttonVariants }
+export { Button, buttonVariants };
```

ESLint

[error] Fast refresh only works when a file only exports components. Use a new file to share constants or functions between components.

(react-refresh/only-export-components)